compiler-plugin/src/main/kotlin/arrow/meta/quotes/Quote.kt

```diff
@@ -327,16 +350,24 @@ inline fun <reified K : KtElement> processContext(source: KtFile, match: K.() ->
     K::class.java.isAssignableFrom(it.javaClass)
 }.firstOrNull { (it as K).match() } as K
 
-fun java.util.ArrayList<KtFile>.replaceFiles(file: KtFile, newFile: KtFile) {
+fun java.util.ArrayList<KtFile>.replaceFiles(file: KtFile, newFile: List<KtFile>) {
   val fileIndex = indexOf(file)
   removeAt(fileIndex)
-  add(fileIndex, newFile)
+  addAll(fileIndex, newFile)
 }
 
-fun CompilerContext.changeSource(file: KtFile, newSource: String): KtFile =
-  cli {
+fun CompilerContext.changeSource(file: KtFile, newSource: String, rootFile: KtFile): KtFile {
+  var virtualFile = rootFile.virtualFile
+  if (file.name != DEFAULT_META_FILE_NAME) {
+      val directory = File("build/generated/source/kapt/main")
```

Comment:
Thanks again Daniel for your contribution!! :tada: Please, this path just will work on UNIX platforms. And I don't know if the result is the expected one. For example, the result for the first test:

```
compiler-plugin/build/generated/
└── source
    ├── kapt
    │   ├── main
    │   │   └── NewSource_Generated.kt
    │   └── test
    └── kaptKotlin
        ├── main
        └── test
```